web/src/containers/ActivityDetailGoals.js

```diff
@@ -1,16 +1,76 @@
-import React from 'react';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
 import { connect } from 'react-redux';
 
+import {
+  addActivityGoal as addActivityGoalAction,
+  updateActivity as updateActivityAction
+} from '../actions/activities';
 import Collapsible from '../components/Collapsible';
+import { Textarea } from '../components/Inputs2';
 
-const ActivityDetailDescription = () => (
-  <Collapsible title="Needs and Objectives">
-    <div>...</div>
-  </Collapsible>
-);
+class ActivityDetailGoals extends Component {
+  handleChange = (idx, key) => e => {
+    const { value } = e.target;
+    const { activity, updateActivity } = this.props;
+
+    const updates = { goals: { [idx]: { [key]: value } } };
+    updateActivity(activity.id, updates);
+  };
+
+  render() {
+    const { activity, addActivityGoal } = this.props;
+
+    return (
+      <Collapsible title="Needs and Objectives" open>
+        <div className="mb2">
+          List the goals you’re hoping to accomplish as part of this activity:
+        </div>
+        {activity.goals.map((d, i) => (
+          <div key={i} className="mb3">
+            <Textarea
+              name={`goal-${i}`}
+              label={`Goal #${i + 1}`}
+              rows="5"
+              value={d.desc}
+              onChange={this.handleChange(i, 'desc')}
```

Comment:
Should these keys map to the keys in the object going back and forth with the API?

Reply:
possibly, though using the array index may be more efficient - O(1) lookup vs. O(n) using `find`, though if we structure the goals as an object vs an array of objects, then it would be the same...

Reply:
Naw, I just meant `description` instead of `desc`.  😜